include/deal.II/base/utilities.h

```diff
@@ -1468,6 +1486,13 @@ namespace Utilities
          const std::vector<char>::const_iterator &cend,
          const bool                               allow_compression)
   {
+    // See if the object is empty. If so, shortcut the whole
+    // operation and return an empty object. This avoids
+    // dereferencing iterators if the std::vector<char> is empty,
+    // which would not be allowed.
+    if (cbegin == cend)
+      return {};
```

Comment:
This does not strike me as right. We should only ever have no data to unpack if `T` has size zero. Can you guard that case here by `Assert (sizeof(T)==0, ExcInternalError())` or some such?